serif_compose/src/jvmMain/kotlin/xyz/room409/serif/serif_compose/UiPlatform.kt

```diff
@@ -53,28 +55,120 @@ actual object UiPlatform {
-                    TextField(
-                        value = reason,
-                        onValueChange = { reason = it })
-                    Row() {
-                        Button(onClick = { sendRedaction(message.id); close_deletion_dialog(); reason = "" }) {
-                            Text("Confirm")
-                        }
-                        Button(onClick = { close_deletion_dialog(); reason = "" }) {
-                            Text("Cancel")
-                        }
+    var reason by remember { mutableStateOf("") }
+    Dialog(
+        onCloseRequest = { close_deletion_dialog(); reason = "" },
+        state = rememberDialogState(position = WindowPosition(Alignment.Center))
+    ) {
+            Column() {
+                Text("Deleting: ${message.message}")
+                Text("Reason:")
+                TextField(
+                    value = reason,
+                    onValueChange = { reason = it })
+                Row() {
+                    Button(onClick = { sendRedaction(message.id); close_deletion_dialog(); reason = "" }) {
+                        Text("Confirm")
                     }
+                    Button(onClick = { close_deletion_dialog(); reason = "" }) {
+                        Text("Cancel")
+                    }
+                }
+            }
+        }
+}
+
+actual object AudioPlayer {
+    var url = ""
+    var audioPlayer_proc : Process? = null;
+    actual fun loadAudio(audio_url: String) {
+        if(audio_url != url) {
+            // reload vlc process
+            restart_vlc(audio_url);
+            send_command("pause")
+
+            //Update URL
+            url = audio_url
+        }
+    }
+    actual fun play() {
+        send_command("pause")
+    }
+    actual fun isPlaying(): Boolean {
+        if(audioPlayer_proc != null) {
+            val writer = BufferedWriter(OutputStreamWriter(audioPlayer_proc?.getOutputStream()))
+            val reader = BufferedReader(InputStreamReader(audioPlayer_proc?.getInputStream()))
+            /* NOTE: We can't just call readline on reader because vlc is not
+             * guaranteed to print out a new line in stdout while it is waiting
+             * for user input after the '>'. I wasn't able to get the input working
+             * using a cleaner method so for now we loop to clear out anything in
+             * the buffer from previous commands/startup, send the status command,
+             * and then read in the output.
+             *
+             * A better (future) solution would probably be to have vlc open up a local socket
+             * and talk to it over that.
+             */
+            //Toss out old stdout data
+            while(reader.ready()) {
+                if(reader.read().toChar() == '>')  break
+            }
+
+            //Send status command and wait for it to process
+            writer.write("status\n")
+            writer.flush()
+            Thread.sleep(5)
```

Comment:
I'm a bit worried about this sleep, is there a way we can omit it?

Reply:
Possibly. Currently the communication with the spawned process is done on the main thread so we can't sit forever waiting but we need to give the process some time to come up with a result for that "status" command we send it. We could probably swap to isPlaying take in a callback, do this status command on a separate thread instead.

Reply:
Updated PR